Week4/week4.py

```diff
@@ -0,0 +1,165 @@
+    city=input("Enter city: ")
+    L=[fname,lname,contact,email,city]
+    conn.execute("INSERT INTO CONTACT VALUES(?,?,?,?,?);",L)
+    print('\n\nContact inserted sucessfully.\n')
+    con.commit()
+
+# updating record 
+
+def updateContacts(con):
+    cur=con.cursor()
+    name_search=input("\n\nEnter their First name: ")
+    new_contact=input("Enter New Contact No :")
+    cur.execute(f"update CONTACT set contact='{new_contact}' where fname='{name_search}'")
+    print("\n\nContact updated successfully.\n")
+    con.commit()
+
+#deleting record 
+
+def deleteContacts(con):
+    cur=con.cursor()
+    name_search=input("\n\nEnter their First Name: ")
+    cur.execute(f"delete from CONTACT where fname='{name_search}'")
+    print("\n\nContact deleted successfully.\n")
+    con.commit()
+
+# searching record
+
+def searchContacts(con):
+    cur=con.cursor()
+    name_search=input("\n\nEnter their First Name: ")
+    cur.execute(f"select * from CONTACT where fname='{name_search}'")
+    records=cur.fetchall()
+    print('\n____________________________________________________________________________________')
+    print('Fname\tLaname\tContact\t\tEmail\t\t\tCity')
+    print('______________________________________________________________________________________')
+    for rows in records:
+        print('{}\t{}\t{}\t{}\t{}'.format(rows[0],rows[1],rows[2],rows[3],rows[4]))     
+    print('\n\nThis are all available records\n')
+    con.commit()
+
+# def viewRecords(con):
+    cur=con.cursor()
+    cur.execute('select * from CONTACT')
+    records=cur.fetchall()
+    print('\n____________________________________________________________________________________')
+    print('Fname\tLaname\tContact\t\tEmail\t\t\tCity')
+    print('______________________________________________________________________________________')
+    for rows in records:
+        print('{}\t{}\t{}\t{}\t{}'.format(rows[0],rows[1],rows[2],rows[3],rows[4]))       
+    print('\n\nThis are all available records')
+    con.commit()
+
+# operation function 
+
+def operationFunct():
+    con=createConnect()
+    tablecreate(con)
+    createLog(con)
+    createTrigger(con)
+    logTrigger(con)
+    choice=1
+    while choice!=0:
+        print('\n-------------------------------------------------------------------------------------------------')
+        print('1- Insert contacts')
+        print('2- Update contacts')
+        print('3- Delete contacts')
+        print('4- Search contacts')
+        print('5- View all records')
+        print('0- Exit the program')
+        choice=int(input('\nEnter your choice: '))     
+        if choice==1:
+            insertRecords(con)      
+        elif choice==2:
+            updateContacts(con)     
+        elif choice==3:
+            deleteContacts(con)     
+        elif choice==4:
+            searchContacts(con)     
+        elif choice==5:
+            viewRecords(con)       
+    con.close()
+import sqlite3 as sq     

```

Comment:
why we use import after close connection?

Reply:
To disconnect Database connection, use close() method. If the connection to a database is closed by the user with the close() method, any outstanding transactions are rolled back by the DB.